client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java

```diff
@@ -530,6 +535,91 @@ public void onFailure(Exception e) {
         assertTrue(latch.await(30L, TimeUnit.SECONDS));
     }
 
+    public void testDeletePolicyFromIndex() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+        // setup policy for index
+        Map<String, Phase> phases = new HashMap<>();
+        phases.put("delete", new Phase("delete", TimeValue.timeValueHours(10L),
+            Collections.singletonMap(DeleteAction.NAME, new DeleteAction())));
+        LifecyclePolicy policy = new LifecyclePolicy("my_policy", phases);
+        PutLifecyclePolicyRequest putRequest = new PutLifecyclePolicyRequest(policy);
+        client.indexLifecycle().putLifecyclePolicy(putRequest, RequestOptions.DEFAULT);
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest("my_index",
+            Settings.builder()
+                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+                .put("index.lifecycle.name", "my_policy")
+                .build());
+        client.indices().create(createIndexRequest, RequestOptions.DEFAULT);
+        assertBusy(() -> assertTrue(client.indexLifecycle()
+            .explainLifecycle(new ExplainLifecycleRequest().indices("my_index"), RequestOptions.DEFAULT)
+            .getIndexResponses().get("my_index").managedByILM()));
+
+        // tag::ilm-delete-lifecycle-policy-from-index-request
+        List<String> indices = new ArrayList<>();
+        indices.add("my_index");
+        RemoveIndexLifecyclePolicyRequest request =
+            new RemoveIndexLifecyclePolicyRequest(indices); // <1>
+        // end::ilm-delete-lifecycle-policy-from-index-request
+
+
+        // tag::ilm-delete-lifecycle-policy-from-index-execute
+        RemoveIndexLifecyclePolicyResponse response = client
+            .indexLifecycle()
+            .removeIndexLifecyclePolicy(request, RequestOptions.DEFAULT);
+        // end::ilm-delete-lifecycle-policy-from-index-execute
+
+        // tag::ilm-delete-lifecycle-policy-from-index-response
+        boolean hasFailures = response.hasFailures(); // <1>
+        List<String> failedIndexes = response.getFailedIndexes(); // <2>
+        // end::ilm-delete-lifecycle-policy-from-index-response
+
+        {
+            assertFalse(hasFailures);
+            Map<String, Object> indexSettings = getIndexSettings("my_index");
+            assertTrue(Strings.isNullOrEmpty((String) indexSettings.get("index.lifecycle.name")));
+        }
+
+        // re-apply policy on index
+        updateIndexSettings("my_index", Settings.builder().put("index.lifecycle.name", "my_policy"));
+        assertBusy(() -> assertTrue(client.indexLifecycle()
```

Comment:
I expect this to fail until https://github.com/elastic/elasticsearch/pull/35800 is merged

Reply:
was merged, looks like this is passing now